### PR TITLE
MEPTS-18 Implement mechanism for partners to manually set concept ids used by MER indicators

### PR DESCRIPTION
**Pull request by ynurmahomed**
Use global properties for metadata uuids and display alert if metadata fails to load correctly.

### CODE REVIEW

File: omod/src/main/resources/config.xml [hunk 1]
```diff
@@ -53,17 +53,85 @@
...
+			UUID for the INICIAR concept
 		</description>
 	</globalProperty>
-	/Required Global Properties -->
+
+    <globalProperty>
+        <property>@MODULE_ID@.historicalStartDateConceptUuid</property>
+        <defaultValue>e1d8f690-1d5f-11e0-b929-000c29ad1d07</defaultValue>
+        <description>
+            UUID for the DATA DE INICIO DO TARV concept
```
Comment (ynurmahomed) [inline omod/src/main/resources/config.xml]: Used names in Portuguese for the descriptions.

File: api/src/main/java/org/openmrs/module/eptsreports/EptsReportsActivator.java [hunk 2]
```diff
@@ -54,7 +55,13 @@ public void willStop() {
 	 * @see #started()
 	 */
 	public void started() {
-		reportsInitializer.initializeReports();
+		try {
+			reportsInitializer.initializeReports();
+		}
+		catch (IllegalArgumentException e) {
```
Comment (ynurmahomed) [inline api/src/main/java/org/openmrs/module/eptsreports/EptsReportsActivator.java]: I think `IllegalArgumentException` here does not necessarily mean that a metadata lookup failed.

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/EptsReportsActivator.java]: Yeah, it means there was no resource based on the look up identifier given

File: omod/src/main/resources/config.xml (hunk 1 above)
Comment (ningosi) [reply omod/src/main/resources/config.xml]: I think in code we can stick to english except those that will appear on the UI for the user to go through. We effect multiple locales through messages.properties files in the config

Comment (ynurmahomed) [reply omod/src/main/resources/config.xml]: I've used names in pt to make it easier to search for the concept, no need to change the system language. This description only appears in the UI.

File: api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java [hunk 3]
```diff
@@ -17,89 +17,81 @@
...
-		return getEncounterType(S_TARV_FARMACIA);
+		String uuid = Context.getAdministrationService().getGlobalProperty(EptsReportConstants.GLOBAL_PROPERTY_S_TARV_FARMACIA_ENCOUNTER_TYPE_UUID);
+		return getEncounterType(uuid);
 	}
 	
 	public EncounterType getMisauLaboratorioEncounterType() {
-		return getEncounterType(MISAU_LABORATORIO);
-	}
-	
-	public Concept getDateOfDelivery() {
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java]: @ynurmahomed why did you remove these concepts?

File: api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java [hunk 4]
```diff
@@ -17,89 +17,81 @@
...
+	// Encounter types
 	public EncounterType getAdultoSeguimentoEncounterType() {
-		return getEncounterType(S_TARV_ADULTO_SEGUIMENTO);
+		String uuid = Context.getAdministrationService().getGlobalProperty(EptsReportConstants.GLOBAL_PROPERTY_S_TARV_ADULTO_SEGUIMENTO_ENCOUNTER_TYPE_UUID); // encounterType_id=6
+		return getEncounterType(uuid);
 	}
 	
 	public EncounterType getARVPediatriaSeguimentoEncounterType() {
-		return getEncounterType(S_TARV_PEDIATRIA_SEGUIMENTO);
+		String uuid = Context.getAdministrationService().getGlobalProperty(EptsReportConstants.GLOBAL_PROPERTY_S_TARV_PEDIATRIA_SEGUIMENTO_ENCOUNTER_TYPE_UUID); // encounterType_id
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java]: @ynurmahomed please move the commenting to above the function.

File: api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java [hunk 5]
```diff
@@ -17,89 +17,81 @@
...
 
 @Component("hivMetadata")
 public class HivMetadata extends CommonMetadata {
 	
-	// ECOUNTER TYPES
-	private String S_TARV_ADULTO_SEGUIMENTO = "e278f956-1d5f-11e0-b929-000c29ad1d07"; // encounterType_id=6
-	
-	private String S_TARV_PEDIATRIA_SEGUIMENTO = "e278fce4-1d5f-11e0-b929-000c29ad1d07"; // encounterType_id = 9
-	
-	private String MISAU_LABORATORIO = "e2790f68-1d5f-11e0-b929-000c29ad1d07"; // encounterType_id = 13
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java]: Please move these comments for the encoutnerType_id to above the relevant function

File: api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java (hunk 3 above)
Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/metadata/HivMetadata.java]: @nthfloor the concepts moved to L48 and L78.

File: api/src/main/java/org/openmrs/module/eptsreports/EptsReportsActivator.java [hunk 6]
```diff
@@ -58,7 +59,7 @@ public void started() {
 		try {
 			reportsInitializer.initializeReports();
 		}
-		catch (IllegalArgumentException e) {
+		catch (MetadataLookupException e) {
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/EptsReportsActivator.java]: @ynurmahomed this looks good :+1: 

Only suggestion I have is to also add a catch for a generic exception to print a general message in the event of any other kind of error.
